adal/src/main/java/com/microsoft/aad/adal/AuthenticationResult.java

```diff
@@ -282,11 +318,15 @@ public String getErrorLogInfo() {
      * @return true if expired
      */
     public boolean isExpired() {
```

Comment:
mIsExtendedLifeTimeToken will be set to true when outage flag is true and we do the retry. At this time, the expiration will be checked based on the ExtendedExpiresOn. Otherwise, mIsExtendedLifeTimeToken will be false and we check if the result is expired based on the ExpiresOn.